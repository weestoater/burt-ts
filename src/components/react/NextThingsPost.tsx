import React from "react";

import NextJSLogo from "../../assets/nextjs-logo.png";

export default function NextThingsPost() {
  return (
    <div className="col-md-6 col-sm-6 mb-4">
      <div className="card">
        <div className="card-header">
          <i className="bi bi-binoculars" />
          NextJS things Next...
        </div>
        <div className="card-body">
          <img src={NextJSLogo} alt="NextJS logo" className="circle-right" />
          <p>
            On a recent works 'global hackathon' I did a small PoC using NextJS
            and Bootstrap. I really wasn't sure what to expect with NextJS, but
            somethings were really interesting and others very confusing /
            annoying.
          </p>
          <p>
            The way that NextJs looks after the '<i>server-side</i>'{" "}
            <code>api</code> is very neat - and extensible. I was easily able to
            hook up my data sets and get the PoC showing actual data. This was
            one of the main issues I was aiming to overcome, so such a quick-win
            was really appreciated.
          </p>

          <p>
            Integrating <b>agCharts</b> to show the data was also a breeze,
            because of the support for the ReactJS libraries. Once again, the
            data was being rendered visually really swiftly and I was flying
            through my todo list.
          </p>

          <p>
            Integrating <b>Bootstrap</b> libraries on the other hand was{" "}
            <b>not</b> intuitive / as easy as I had expected it to be. There was
            a <b>lot more googling</b> and <b>stackoverflow</b> searching
            required to find a solution that DID work, rather than the 5 mins it
            takes to implement in a ReactJS project.
          </p>

          <p>
            Overall, I'm not sure how I feel about the NextJS framework, but I
            know several people who swear by it - and they are people I trust
            and respect - so I am probably just being picky. A couple of good
            pluralsight tutorials should probably get me through the learning
            curve and over the top of whatever issues I'm likely to encounter.{" "}
          </p>

          <p>Wish me luck - I'm going to need it.</p>
        </div>
      </div>
    </div>
  );
}
