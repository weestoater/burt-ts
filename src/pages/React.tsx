import React from "react";

import RandomUsersPost from "../components/react/RandomUserPost";
import ReactHooksPost from "../components/react/ReactHooksPost";
import NextThingsPost from "../components/react/NextThingsPost";

export default function ReactPage() {
  return (
    <>
      <div className="row">
        <div className="col-sm-12">
          <h1>React</h1>
        </div>
        <NextThingsPost />
        <ReactHooksPost />
        <RandomUsersPost />
      </div>
    </>
  );
}
