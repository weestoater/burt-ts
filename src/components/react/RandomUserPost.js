import React, {useState, useEffect} from 'react';
import getApi from '../../adapters/commonApi';
import Spinner from '../general/Spinner';
import UserItem from './UserItem';

export default function RandomUsersPost() {

    const [users, setUsers] = useState({});
    const apiUrl = "https://randomuser.me/api/?results=3";

    useEffect(() => {
        getApi(apiUrl).then(setUsers);
      }, []);

    return (
        <div className="col-md-6 col-sm-12 mb-4">
            <div className="card">
                <div className="card-header">
                    <i className="bi bi-filetype-jsx" />
                    Random Users API
                </div>
                <div className="card-body users">
                {users.results ? (
                    users.results.map((user, key) => <UserItem key={key} item={user} /> )
                ) : ( 
                <Spinner /> 
                )}
                </div>
            </div>
        </div>
    );
}
