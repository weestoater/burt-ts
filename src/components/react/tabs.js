import React, { useState, useCallback } from "react";

function Tabs({ children }) {
  const initialTab = children[0].props.label
  const [activeTab, setActiveTab] = useState(initialTab)
  const handleActiveTab = useCallback(label => setActiveTab(label), [])

  const tabs = children.map(child => (
    <button tabIndex={"0"}
      onClick={e => {
        e.preventDefault();
        handleActiveTab(child.props.label);
      }}
      className={
        child.props.label === activeTab
          ? ["tabs__tab", "tabs__tab-active"].join(" ")
          : "tabs__tab"
      }
      key={child.props.label}
    >
      {child.props.tabName}
    </button>
  ))
  
  const tabContent = children.filter(child => child.props.label === activeTab)

  return (
    <div>
      <div className="tabs__box">{tabs}</div>
      <div>{tabContent}</div>
    </div>
  )
}

function Tab(props) {
	return <>{props.children}</>
}

export { Tabs, Tab }