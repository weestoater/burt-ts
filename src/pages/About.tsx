import React from "react";
import AboutContent from "../components/about/AboutContent";
import BioContent from "../components/about/BioContent";

export default function About() {
  return (
    <>
      <div className="row">
        <div className="col-sm-12">
          <h1 data-testid="pagetitle">About</h1>
        </div>
        <div className="col-lg-6 col-md-6 col-sm-12">
          <AboutContent />
        </div>
        <div className="col-lg-6 col-md-6 col-sm-12">
          <BioContent />
        </div>
      </div>
    </>
  );
}
