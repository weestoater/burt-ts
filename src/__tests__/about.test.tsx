import React from "react";
import { render, screen } from "@testing-library/react";
import About from "../pages/About";

test("renders a the about page", () => {
  render(<About />);
  const headerText = screen.getByTestId("pagetitle");
  expect(headerText).toBeInTheDocument();
});

test("renders what is burtware.com heading", () => {
  render(<About />);
  const headerText = screen.getByText(/what is burtware.com/i);
  expect(headerText).toBeInTheDocument();
});

test("renders who is burtware heading", () => {
  render(<About />);
  const headerText = screen.getByText(/who is burtware/i);
  expect(headerText).toBeInTheDocument();
});

test("should match snapshot", () => {
  const { container } = render(<About />);
  expect(container).toMatchSnapshot();
});
