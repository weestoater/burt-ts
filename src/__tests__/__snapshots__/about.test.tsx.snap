// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-sm-12"
    >
      <h1
        data-testid="pagetitle"
      >
        About
      </h1>
    </div>
    <div
      class="col-sm-6"
    >
      <div
        class="card mb-4"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-wrench-adjustable-circle"
          />
          what is burtware.com ?
        </div>
        <div
          class="card-body"
        >
          <p>
            <img
              alt="screenshot of VS Code editor with this page of code on view"
              class="img-thumbnail"
              src="vs-code.png"
            />
          </p>
          <p>
            burtware is the nickname I've used for my personal 'playgrounds' online since 1999, when I first cut my teeth in 
            <code>
              html
            </code>
             
            & 
            <code>
              design
            </code>
            . Since then I've gone through a number of different efforts.
          </p>
          <ul>
            <li>
              A hand rolled CMS on php
            </li>
            <li>
              Numerous versions of WordPress
            </li>
            <li>
              An umbraco site - didn't last long
            </li>
            <li>
              A few Angular / AngularJS versions
            </li>
            <li>
              A couple of React sites
            </li>
          </ul>
          <p>
            This current itteration is 
            <code>
              ReactJS
            </code>
            , 
            <code>
              SASS
            </code>
             and hosted on 
            <code>
              AWS
            </code>
            . It is a site for me to play with and learn more about 
            <code>
              ReactJS
            </code>
             and 
            <code>
              AWS
            </code>
            , as I use both technologies in my day job, as well as Angular; Node; Cypress and multiple CSS frameworks.
          </p>
          <p>
            I also want to use it to trial different experiments and implement the fabulous React Testing Library based on the amazing
             
            <a
              href="http://www.testingjavascript.com"
              rel="noreferrer"
              target="_blank"
            >
              testing javascript
            </a>
             
            course from 
            <b>
              Kent C. Dodds
            </b>
            . I work very heavily in Accessibility and I would like to flex some testing muscles too.
          </p>
        </div>
      </div>
    </div>
    <div
      class="col-sm-6"
    >
      <div
        class="card mb-4"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-wrench-adjustable-circle"
          />
           who is burtware ?
        </div>
        <div
          class="card-body"
        >
          <img
            alt="avatar of burtware aka Ian Burrett"
            class="float-end img-thumbnail ms-3 mb-3"
            src="burt.png"
          />
          <p>
            I am Ian Burrett, a Scottish Lead UI Engineer / Web Developer, working in Glasgow, UK and living just outside the fine city.
          </p>
          <p>
            I'm a father of two amazing kids and a husband to my 
            <b>
              very
            </b>
             long suffering wife. We have an adorable dog called Buster and he is the absolutely best boy ever - spoilt and loved to bits.
          </p>
          <p
            class="text-center"
          >
            <img
              alt=""
              class="img-thumbnail"
              src="buster.jpg"
            />
          </p>
          <p>
            When not working or ferrying my kids to one of their many clubs, I like to tinker with code / websites; listen to 'rubbish' music; watch weird and wonderful stuff on various streaming services. I occassionaly make it along to see Motherwell FC too and have been know to strum a tune on the guitar once in a blue moon.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
